## 기후별 교통사고 현황
- 일단 월별로 나누지 않고 최근 5년 데이터로 사고건수, 사망자 합해서 비율 따져볼 예정
- 사망률 = 사망자/사고건수 또는 사망률 = 사망자/전체인구
- 아니다 싶으면 월별로 나눠서 최근 2년 데이터로 진행
<br>
<br>

#### 데이터 사용 장단점
1. 최근 5년 전체 데이터 사용
- 장점 : 최근 5년간의 비율을 한눈에 볼 수 있어서 비교하기 좋음
- 단점 : 월별로 구분하기엔 데이터가 너무 많아 그래프 나타내기엔 무리

2. 최근 2년 월별 데이터 사용
- 장점 : 전체보다 좀 더 세세하게 볼 수 있어서 월별로 비교하기엔 좋음
- 단점 : 연도별로 비교할 수가 없고 눈이나 안개 같은 경우엔 값이 없는 경우도 있음,여름이나 겨울 같은 경우 비나 눈이 너무 몰려있는 느낌

=> 둘다 해서 보여주는 것도 좋을 것 같음

In [3]:
#필요한 라이브러리
import pandas as pd
import numpy as np

In [4]:
#데이터 불러오기(월별로 말고 일단 전체로 취합)
W_R_TA = pd.read_csv('C:/Users/ankm6/DS_Project/도로종류별_기상상태별_교통사고_최근5년 전체데이터.csv', encoding='cp949', header=1)
W_R_TA

도로종류별(1) 기상상태별(1) 월별(1)  사고건수 (건)  사망자수 (명)  부상자수 (명)  사고건수 (건).1  \
0      일반국도       맑음    전체     15492       597     26115       15872   
1      일반국도       흐림    전체       694        56      1157         702   
2      일반국도        비    전체      1021        56      1801        1325   
3      일반국도       안개    전체        42         5        79          74   
4      일반국도        눈    전체       168         3       313         254   
5      일반국도    기타/불명    전체       102         5       163         114   
6       지방도       맑음    전체     14905       484     22859       13744   
7       지방도       흐림    전체       712        38      1160         662   
8       지방도        비    전체       830        38      1373        1024   
9       지방도       안개    전체        40         4        64          49   
10      지방도        눈    전체       152         6       243         202   
11      지방도    기타/불명    전체       133         1       193          92   
12   특별광역시도       맑음    전체     79967       849    114062       77482   
13   특별광역시도       흐림    전체      2920        63      4113        2756   
14   특별광역시도        비    전체      4443        58      6600        5700   
15   특별광역시도       안개    전체        22         1        37          18   
16   특별광역시도        눈    전체       438         6       672         483   
17   특별광역시도    기타/불명    전체       441         5       600         544   
18       시도       맑음    전체     67170      1055     99203       65546   
19       시도       흐림    전체      2185        62      3399        2194   
20       시도        비    전체      3783        74      5816        5088   
21       시도       안개    전체        55         3        92          93   
22       시도        눈    전체       437        12       743         524   
23       시도    기타/불명    전체       451         4       724         288   
24       군도       맑음    전체      7636       260     10994        6665   
25       군도       흐림    전체       318        19       517         360   
26       군도        비    전체       377        17       541         461   
27       군도       안개    전체        18         2        23          21   
28       군도        눈    전체        59         3        86          79   
29       군도    기타/불명    전체        17         0        19          30   
30     고속국도       맑음    전체      3613       195      8573        3459   
31     고속국도       흐림    전체       183        17       396         157   
32     고속국도        비    전체       281        30       634         394   
33     고속국도       안개    전체         5         1        22          12   
34     고속국도        눈    전체        60         5       148          51   
35     고속국도    기타/불명    전체         4         0         5           6   
36    기타/불명       맑음    전체      6526       135      8450        9372   
37    기타/불명       흐림    전체       213         7       276         416   
38    기타/불명        비    전체       284         5       389         553   
39    기타/불명       안개    전체         5         1         6          15   
40    기타/불명        눈    전체        29         2        35          70   
41    기타/불명    기타/불명    전체       104         1       134         197   

    사망자수 (명).1  부상자수 (명).1  사고건수 (건).2  사망자수 (명).2  부상자수 (명).2  사고건수 (건).3  \
0          512       26268       17908         471       29873       16454   
1           37        1232         652          32        1093         804   
2           65        2248        1344          30        2239        1477   
3            9         134          59           3          91          44   
4            8         467          70           0         152          83   
5            4         185         133           5         194         183   
6          418       21609       14957         394       23630       12469   
7           35        1032         568          34         901         556   
8           47        1642        1083          27        1666        1078   
9            5          79          42           2          69          27   
10           5         374          40          

In [5]:
df = pd.DataFrame(W_R_TA)
df

도로종류별(1) 기상상태별(1) 월별(1)  사고건수 (건)  사망자수 (명)  부상자수 (명)  사고건수 (건).1  \
0      일반국도       맑음    전체     15492       597     26115       15872   
1      일반국도       흐림    전체       694        56      1157         702   
2      일반국도        비    전체      1021        56      1801        1325   
3      일반국도       안개    전체        42         5        79          74   
4      일반국도        눈    전체       168         3       313         254   
5      일반국도    기타/불명    전체       102         5       163         114   
6       지방도       맑음    전체     14905       484     22859       13744   
7       지방도       흐림    전체       712        38      1160         662   
8       지방도        비    전체       830        38      1373        1024   
9       지방도       안개    전체        40         4        64          49   
10      지방도        눈    전체       152         6       243         202   
11      지방도    기타/불명    전체       133         1       193          92   
12   특별광역시도       맑음    전체     79967       849    114062       77482   
13   특별광역시도       흐림    전체      2920        63      4113        2756   
14   특별광역시도        비    전체      4443        58      6600        5700   
15   특별광역시도       안개    전체        22         1        37          18   
16   특별광역시도        눈    전체       438         6       672         483   
17   특별광역시도    기타/불명    전체       441         5       600         544   
18       시도       맑음    전체     67170      1055     99203       65546   
19       시도       흐림    전체      2185        62      3399        2194   
20       시도        비    전체      3783        74      5816        5088   
21       시도       안개    전체        55         3        92          93   
22       시도        눈    전체       437        12       743         524   
23       시도    기타/불명    전체       451         4       724         288   
24       군도       맑음    전체      7636       260     10994        6665   
25       군도       흐림    전체       318        19       517         360   
26       군도        비    전체       377        17       541         461   
27       군도       안개    전체        18         2        23          21   
28       군도        눈    전체        59         3        86          79   
29       군도    기타/불명    전체        17         0        19          30   
30     고속국도       맑음    전체      3613       195      8573        3459   
31     고속국도       흐림    전체       183        17       396         157   
32     고속국도        비    전체       281        30       634         394   
33     고속국도       안개    전체         5         1        22          12   
34     고속국도        눈    전체        60         5       148          51   
35     고속국도    기타/불명    전체         4         0         5           6   
36    기타/불명       맑음    전체      6526       135      8450        9372   
37    기타/불명       흐림    전체       213         7       276         416   
38    기타/불명        비    전체       284         5       389         553   
39    기타/불명       안개    전체         5         1         6          15   
40    기타/불명        눈    전체        29         2        35          70   
41    기타/불명    기타/불명    전체       104         1       134         197   

    사망자수 (명).1  부상자수 (명).1  사고건수 (건).2  사망자수 (명).2  부상자수 (명).2  사고건수 (건).3  \
0          512       26268       17908         471       29873       16454   
1           37        1232         652          32        1093         804   
2           65        2248        1344          30        2239        1477   
3            9         134          59           3          91          44   
4            8         467          70           0         152          83   
5            4         185         133           5         194         183   
6          418       21609       14957         394       23630       12469   
7           35        1032         568          34         901         556   
8           47        1642        1083          27        1666        1078   
9            5          79          42           2          69          27   
10           5         374          40          

In [6]:
#사고건수만 일단 추출
accident = df[['도로종류별(1)', '기상상태별(1)','월별(1)', '사고건수 (건)','사망자수 (명)', '사고건수 (건).1', '사망자수 (명).1',
               '사고건수 (건).2', '사망자수 (명).2', '사고건수 (건).3', '사망자수 (명).3', '사고건수 (건).4', '사망자수 (명).4']]
accident

도로종류별(1) 기상상태별(1) 월별(1)  사고건수 (건)  사망자수 (명)  사고건수 (건).1  사망자수 (명).1  \
0      일반국도       맑음    전체     15492       597       15872         512   
1      일반국도       흐림    전체       694        56         702          37   
2      일반국도        비    전체      1021        56        1325          65   
3      일반국도       안개    전체        42         5          74           9   
4      일반국도        눈    전체       168         3         254           8   
5      일반국도    기타/불명    전체       102         5         114           4   
6       지방도       맑음    전체     14905       484       13744         418   
7       지방도       흐림    전체       712        38         662          35   
8       지방도        비    전체       830        38        1024          47   
9       지방도       안개    전체        40         4          49           5   
10      지방도        눈    전체       152         6         202           5   
11      지방도    기타/불명    전체       133         1          92           3   
12   특별광역시도       맑음    전체     79967       849       77482         715   
13   특별광역시도       흐림    전체      2920        63        2756          54   
14   특별광역시도        비    전체      4443        58        5700          77   
15   특별광역시도       안개    전체        22         1          18           3   
16   특별광역시도        눈    전체       438         6         483           4   
17   특별광역시도    기타/불명    전체       441         5         544           3   
18       시도       맑음    전체     67170      1055       65546         925   
19       시도       흐림    전체      2185        62        2194          63   
20       시도        비    전체      3783        74        5088          91   
21       시도       안개    전체        55         3          93           7   
22       시도        눈    전체       437        12         524           4   
23       시도    기타/불명    전체       451         4         288           4   
24       군도       맑음    전체      7636       260        6665         231   
25       군도       흐림    전체       318        19         360          17   
26       군도        비    전체       377        17         461          16   
27       군도       안개    전체        18         2          21           1   
28       군도        눈    전체        59         3          79           2   
29       군도    기타/불명    전체        17         0          30           0   
30     고속국도       맑음    전체      3613       195        3459         179   
31     고속국도       흐림    전체       183        17         157          28   
32     고속국도        비    전체       281        30         394          39   
33     고속국도       안개    전체         5         1          12           3   
34     고속국도        눈    전체        60         5          51           3   
35     고속국도    기타/불명    전체         4         0           6           0   
36    기타/불명       맑음    전체      6526       135        9372         133   
37    기타/불명       흐림    전체       213         7         416          12   
38    기타/불명        비    전체       284         5         553          15   
39    기타/불명       안개    전체         5         1          15           1   
40    기타/불명        눈    전체        29         2          70           0   
41    기타/불명    기타/불명    전체       104         1         197           3   

    사고건수 (건).2  사망자수 (명).2  사고건수 (건).3  사망자수 (명).3  사고건수 (건).4  사망자수 (명).4  
0        17908         471       16454         452       16479         427  
1          652          32         804          39         796          33  
2         1344          30        1477          55        1326          49  
3           59           3          44           3          43           6  
4           70           0          83           1         165           4  
5          133           5         183           8         174           5  
6        14957         394       12469         318       12256         271  
7          568          34         556          20         507          26  
8         1083          27        1078          42         911          30  
9           42           2          27           2

In [7]:
#사고건수_열 이름 변경
accident.columns = ['도로','기상','월','사고_2017', '사망_2017', '사고_2018', '사망_2018', '사고_2019', 
                    '사망_2019', '사고_2020', '사망_2020','사고_2021', '사망_2021']

accident.head()

도로  기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
0  일반국도  맑음  전체    15492      597    15872      512    17908      471   
1  일반국도  흐림  전체      694       56      702       37      652       32   
2  일반국도   비  전체     1021       56     1325       65     1344       30   
3  일반국도  안개  전체       42        5       74        9       59        3   
4  일반국도   눈  전체      168        3      254        8       70        0   

   사고_2020  사망_2020  사고_2021  사망_2021  
0    16454      452    16479      427  
1      804       39      796       33  
2     1477       55     1326       49  
3       44        3       43        6  
4       83        1      165        4

In [2]:
pip install population

### 도로 구분 없이 기상별로 데이터 합치기

In [8]:
accident[accident['기상']=='맑음']

도로  기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
0     일반국도  맑음  전체    15492      597    15872      512    17908      471   
6      지방도  맑음  전체    14905      484    13744      418    14957      394   
12  특별광역시도  맑음  전체    79967      849    77482      715    82866      613   
18      시도  맑음  전체    67170     1055    65546      925    67552      840   
24      군도  맑음  전체     7636      260     6665      231     7096      221   
30    고속국도  맑음  전체     3613      195     3459      179     3757      171   
36   기타/불명  맑음  전체     6526      135     9372      133    12355      169   

    사고_2020  사망_2020  사고_2021  사망_2021  
0     16454      452    16479      427  
6     12469      318    12256      271  
12    74248      543    70945      557  
18    61845      746    57822      681  
24     6670      183     6008      174  
30     3489      159     4255      138  
36     9040      108    11148      124

#### 1. 맑은 날씨

In [11]:
sunny = pd.DataFrame(accident[accident['기상']=='맑음'])

In [211]:
#사고 건수
sunny_2017 = sunny['사고_2017'].sum()
sunny_2018 = sunny['사고_2018'].sum()
sunny_2019 = sunny['사고_2019'].sum()
sunny_2020 = sunny['사고_2020'].sum()
sunny_2021 = sunny['사고_2021'].sum()

print(sunny_2017, sunny_2018, sunny_2019, sunny_2020, sunny_2021)

195309 192140 206491 184215 178913


In [210]:
#사망자 수
sunny_die_2017 = sunny['사망_2017'].sum()
sunny_die_2018 = sunny['사망_2018'].sum()
sunny_die_2019 = sunny['사망_2019'].sum()
sunny_die_2020 = sunny['사망_2020'].sum()
sunny_die_2021 = sunny['사망_2021'].sum()

print(sunny_die_2017, sunny_die_2018, sunny_die_2019, sunny_die_2020, sunny_die_2021)

3575 3113 2879 2509 2372


In [115]:
sunny_2017 = (sunny['사망_2017'].sum() / sunny['사고_2017'].sum())*100
sunny_2018 = (sunny['사망_2018'].sum() / sunny['사고_2018'].sum())*100
sunny_2019 = (sunny['사망_2019'].sum() / sunny['사고_2019'].sum())*100
sunny_2020 = (sunny['사망_2020'].sum() / sunny['사고_2020'].sum())*100
sunny_2021 = (sunny['사망_2021'].sum() / sunny['사고_2021'].sum())*100

print(sunny_2017, sunny_2018, sunny_2019, sunny_2020, sunny_2021)

1.8304328013558004 1.6201727906734673 1.3942496283130985 1.3619954943951362 1.3257840402877374


In [156]:
sunny_total = pd.DataFrame({'맑음':[sunny_2017, sunny_2018, sunny_2019, sunny_2020, sunny_2021]})
sunny_total

맑음
0  195309
1  192140
2  206491
3  184215
4  178913

#### 2. 흐린 날씨

In [21]:
accident[accident['기상']=='흐림']

도로  기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
1     일반국도  흐림  전체      694       56      702       37      652       32   
7      지방도  흐림  전체      712       38      662       35      568       34   
13  특별광역시도  흐림  전체     2920       63     2756       54     2622       40   
19      시도  흐림  전체     2185       62     2194       63     2076       47   
25      군도  흐림  전체      318       19      360       17      288       16   
31    고속국도  흐림  전체      183       17      157       28      126       13   
37   기타/불명  흐림  전체      213        7      416       12      426        9   

    사고_2020  사망_2020  사고_2021  사망_2021  
1       804       39      796       33  
7       556       20      507       26  
13     2639       43     2568       37  
19     2349       53     2211       57  
25      346       16      283       16  
31      130       18      147       16  
37      392       15      470       12

In [22]:
cloudy = pd.DataFrame(accident[accident['기상']=='흐림'])

In [157]:
#사고건수 합하기
cloudy_2017 = cloudy['사고_2017'].sum()
cloudy_2018 = cloudy['사고_2018'].sum()
cloudy_2019 = cloudy['사고_2019'].sum()
cloudy_2020 = cloudy['사고_2020'].sum()
cloudy_2021 = cloudy['사고_2021'].sum()

cloudy_2017, cloudy_2018, cloudy_2019, cloudy_2020, cloudy_2021

(7225, 7247, 6758, 7216, 6982)

In [40]:
cloudy_2017 = (cloudy['사망_2017'].sum() / cloudy['사고_2017'].sum())*100
cloudy_2018 = (cloudy['사망_2018'].sum() / cloudy['사고_2018'].sum())*100
cloudy_2019 = (cloudy['사망_2019'].sum() / cloudy['사고_2019'].sum())*100
cloudy_2020 = (cloudy['사망_2020'].sum() / cloudy['사고_2020'].sum())*100
cloudy_2021 = (cloudy['사망_2021'].sum() / cloudy['사고_2021'].sum())*100

print(cloudy_2017, cloudy_2018, cloudy_2019, cloudy_2020, cloudy_2021)

3.6262975778546718 3.394508072305782 2.826279964486534 2.827050997782705 2.8215411057003723


In [158]:
cloudy_total = pd.DataFrame({'흐림':[cloudy_2017, cloudy_2018, cloudy_2019, cloudy_2020, cloudy_2021]})
cloudy_total

흐림
0  7225
1  7247
2  6758
3  7216
4  6982

#### 3. 눈

In [50]:
accident[accident['기상']=='눈']

도로 기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
4     일반국도  눈  전체      168        3      254        8       70        0   
10     지방도  눈  전체      152        6      202        5       40        0   
16  특별광역시도  눈  전체      438        6      483        4       97        0   
22      시도  눈  전체      437       12      524        4      152        4   
28      군도  눈  전체       59        3       79        2       17        1   
34    고속국도  눈  전체       60        5       51        3       15        3   
40   기타/불명  눈  전체       29        2       70        0       26        0   

    사고_2020  사망_2020  사고_2021  사망_2021  
4        83        1      165        4  
10       74        1      138        4  
16      157        0      463        3  
22      209        1      426        6  
28       33        1       53        1  
34       25        7       61        4  
40       34        1       89        1

In [51]:
snow = pd.DataFrame(accident[accident['기상']=='눈'])

In [151]:
snow_2017 = snow['사고_2017'].sum()
snow_2018 = snow['사고_2018'].sum()
snow_2019 = snow['사고_2019'].sum()
snow_2020 = snow['사고_2020'].sum()
snow_2021 = snow['사고_2021'].sum()

snow_2017, snow_2018, snow_2019, snow_2020, snow_2021

(1343, 1663, 417, 615, 1395)

In [52]:
snow_2017 = (snow['사망_2017'].sum() / snow['사고_2017'].sum())*100
snow_2018 = (snow['사망_2018'].sum() / snow['사고_2018'].sum())*100
snow_2019 = (snow['사망_2019'].sum() / snow['사고_2019'].sum())*100
snow_2020 = (snow['사망_2020'].sum() / snow['사고_2020'].sum())*100
snow_2021 = (snow['사망_2021'].sum() / snow['사고_2021'].sum())*100

print(snow_2017, snow_2018, snow_2019, snow_2020, snow_2021)

2.7550260610573343 1.5634395670475045 1.9184652278177456 1.951219512195122 1.6487455197132617


In [159]:
snow_total = pd.DataFrame({'눈':[snow_2017, snow_2018, snow_2019, snow_2020, snow_2021]})
snow_total

눈
0  1343
1  1663
2   417
3   615
4  1395

#### 4. 비

In [55]:
accident[accident['기상']=='비']

도로 기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
2     일반국도  비  전체     1021       56     1325       65     1344       30   
8      지방도  비  전체      830       38     1024       47     1083       27   
14  특별광역시도  비  전체     4443       58     5700       77     5681       64   
20      시도  비  전체     3783       74     5088       91     4824       74   
26      군도  비  전체      377       17      461       16      443       18   
32    고속국도  비  전체      281       30      394       39      314       19   
38   기타/불명  비  전체      284        5      553       15      688       12   

    사고_2020  사망_2020  사고_2021  사망_2021  
2      1477       55     1326       49  
8      1078       42      911       30  
14     6352       65     5286       65  
20     5352       94     4384       71  
26      520       17      421       12  
32      379       37      403       29  
38      590        5      642       11

In [56]:
rain = pd.DataFrame(accident[accident['기상']=='비'])

In [152]:
rain_2017 = rain['사고_2017'].sum()
rain_2018 = rain['사고_2018'].sum()
rain_2019 = rain['사고_2019'].sum()
rain_2020 = rain['사고_2020'].sum()
rain_2021 = rain['사고_2021'].sum()

rain_2017, rain_2018, rain_2019, rain_2020, rain_2021

(11019, 14545, 14377, 15748, 13373)

In [57]:
rain_2017 = (rain['사망_2017'].sum() / rain['사고_2017'].sum())*100
rain_2018 = (rain['사망_2018'].sum() / rain['사고_2018'].sum())*100
rain_2019 = (rain['사망_2019'].sum() / rain['사고_2019'].sum())*100
rain_2020 = (rain['사망_2020'].sum() / rain['사고_2020'].sum())*100
rain_2021 = (rain['사망_2021'].sum() / rain['사고_2021'].sum())*100

print(rain_2017, rain_2018, rain_2019, rain_2020, rain_2021)

2.5229149650603504 2.4063251976624267 1.6971551784099603 2.0002540005080007 1.996560233305915


In [160]:
rain_total = pd.DataFrame({'비':[rain_2017, rain_2018, rain_2019, rain_2020, rain_2021]})
rain_total

비
0  11019
1  14545
2  14377
3  15748
4  13373

#### 5. 안개

In [59]:
accident[accident['기상']=='안개']

도로  기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
3     일반국도  안개  전체       42        5       74        9       59        3   
9      지방도  안개  전체       40        4       49        5       42        2   
15  특별광역시도  안개  전체       22        1       18        3       15        0   
21      시도  안개  전체       55        3       93        7       67        3   
27      군도  안개  전체       18        2       21        1       22        1   
33    고속국도  안개  전체        5        1       12        3        8        0   
39   기타/불명  안개  전체        5        1       15        1       18        0   

    사고_2020  사망_2020  사고_2021  사망_2021  
3        44        3       43        6  
9        27        2       37        4  
15       10        1       14        1  
21       48        2       76        9  
27       21        0       18        0  
33        4        0       10        2  
39       11        0        9        0

In [60]:
fog = pd.DataFrame(accident[accident['기상']=='안개'])

In [153]:
fog_2017 = fog['사고_2017'].sum()
fog_2018 = fog['사고_2018'].sum()
fog_2019 = fog['사고_2019'].sum()
fog_2020 = fog['사고_2020'].sum()
fog_2021 = fog['사고_2021'].sum()

fog_2017, fog_2018, fog_2019, fog_2020, fog_2021

(187, 282, 231, 165, 207)

In [61]:
fog_2017 = (fog['사망_2017'].sum() / fog['사고_2017'].sum())*100
fog_2018 = (fog['사망_2018'].sum() / fog['사고_2018'].sum())*100
fog_2019 = (fog['사망_2019'].sum() / fog['사고_2019'].sum())*100
fog_2020 = (fog['사망_2020'].sum() / fog['사고_2020'].sum())*100
fog_2021 = (fog['사망_2021'].sum() / fog['사고_2021'].sum())*100

print(fog_2017, fog_2018, fog_2019, fog_2020, fog_2021)

9.090909090909092 10.28368794326241 3.896103896103896 4.848484848484849 10.628019323671497


In [161]:
fog_total = pd.DataFrame({'안개':[fog_2017, fog_2018, fog_2019, fog_2020, fog_2021]})
fog_total

안개
0  187
1  282
2  231
3  165
4  207

#### 6. 기타/불명

In [63]:
accident[accident['기상']=='기타/불명']

도로     기상   월  사고_2017  사망_2017  사고_2018  사망_2018  사고_2019  사망_2019  \
5     일반국도  기타/불명  전체      102        5      114        4      133        5   
11     지방도  기타/불명  전체      133        1       92        3      105        5   
17  특별광역시도  기타/불명  전체      441        5      544        3      514        2   
23      시도  기타/불명  전체      451        4      288        4      282        3   
29      군도  기타/불명  전체       17        0       30        0       15        1   
35    고속국도  기타/불명  전체        4        0        6        0        3        0   
41   기타/불명  기타/불명  전체      104        1      197        3      274        2   

    사고_2020  사망_2020  사고_2021  사망_2021  
5       183        8      174        5  
11      122        8      142        6  
17      640        8      829        4  
23      426        3      576       10  
29       55        3       65        5  
35       12        2        7        2  
41      257        1      467        3

In [64]:
unknown = pd.DataFrame(accident[accident['기상']=='기타/불명'])

In [172]:
unknown_2017 = unknown['사고_2017'].sum()
unknown_2018 = unknown['사고_2018'].sum()
unknown_2019 = unknown['사고_2019'].sum()
unknown_2020 = unknown['사고_2020'].sum()
unknown_2021 = unknown['사고_2021'].sum()

unknown_2017, unknown_2018, unknown_2019, unknown_2020, unknown_2021

(125200, 1271, 1326, 1695, 2260)

In [65]:
unknown_2017 = (unknown['사망_2017'].sum() / unknown['사고_2017'].sum())*100
unknown_2018 = (unknown['사망_2018'].sum() / unknown['사고_2018'].sum())*100
unknown_2019 = (unknown['사망_2019'].sum() / unknown['사고_2019'].sum())*100
unknown_2020 = (unknown['사망_2020'].sum() / unknown['사고_2020'].sum())*100
unknown_2021 = (unknown['사망_2021'].sum() / unknown['사고_2021'].sum())*100

print(unknown_2017, unknown_2018, unknown_2019, unknown_2020, unknown_2021)

1.2779552715654952 1.3375295043273014 1.3574660633484164 1.9469026548672566 1.5486725663716814


In [162]:
unknown_total = pd.DataFrame({'기타':[unknown_2017, unknown_2018, unknown_2019, unknown_2020, unknown_2021]})
unknown_total

기타
0  1252
1  1271
2  1326
3  1695
4  2260

#### 최근 5년치 날씨별 사고 데이터

In [163]:
weather = pd.concat([sunny_total, cloudy_total,snow_total, rain_total, fog_total, unknown_total], axis=1)
weather.index = ['2017','2018','2019','2020', '2021']
weather

맑음    흐림     눈      비   안개    기타
2017  195309  7225  1343  11019  187  1252
2018  192140  7247  1663  14545  282  1271
2019  206491  6758   417  14377  231  1326
2020  184215  7216   615  15748  165  1695
2021  178913  6982  1395  13373  207  2260

맑음 : 일평균 운량이 2.5 미만
흐림 : 일평균 운량이 7.5 이상
비 : 강수량이 0.1㎜ 이상의 날

#막대그래프로 5년치 비교 => 날씨가 어떨때 대체적으로 사고가 많이 나는지
#왜 그런지
#비나 눈 오는날에 사고가 많이 날 것 같지만 사실 맑은날에 훨씬 더 높다~
#날씨별 일수 => 시간대별로 지역별로 다름
#계절별로 등급 나누기 - 봄(3,4,5월), 장마철(6,7월), 여름(8,9월), 11~1월